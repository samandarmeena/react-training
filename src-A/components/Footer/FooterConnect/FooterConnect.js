import React from 'react';
import PropTypes from 'prop-types';
import styles from './FooterConnect.module.scss';
import Text from '../../Shared/Text';
import Image from '../../Shared/Image';

/**
 * FooterConnect component
 * renders Footer connnect container
 * @param {Object} accessibilityImg
 * @param {Array} socialImgs
 * @return Element
 */
const FooterConnect = (props) => {
  const { accessibilityImg, socialImgs } = props;
  return (
    <>
      <div className={styles.footerConnectTitle}>
        <Text text="Join us on" />
      </div>
      <div className={styles.footerConnectLinks}>
        <div className={styles.footerConnectLeft}>
          {socialImgs.map((img, index) => {
            return (
              <span key={index}>
                <Image src={img.src} alt={img.alt} />
              </span>
            );
          })}
        </div>
        <div className={styles.footerConnectRight}>
          <Image src={accessibilityImg.src} alt={accessibilityImg.alt} />
        </div>
      </div>
    </>
  );
};

// proptype validations for props
FooterConnect.propTypes = {
  accessibilityImg: PropTypes.object.isRequired,
  socialImgs: PropTypes.array.isRequired
};

export default FooterConnect;
