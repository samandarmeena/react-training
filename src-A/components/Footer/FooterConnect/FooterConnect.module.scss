.footerConnectLinks {
  display: flex;
  justify-content: space-between;
  .footerConnectLeft {
    display: flex;
    img {
      margin-right: 5px;
    }
  }
  .footerConnectRight {
    img {
      width: 240px;
    }
  }
}
.footerConnectTitle {
  margin-left: 5px;
  margin-bottom: 5px;
}
