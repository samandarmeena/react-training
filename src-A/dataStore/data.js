import { IMAGE_URL } from '../constants/constants';
// Footer links array for FooterLinks component
const footerLinkStack = [
  [
    { headerTitle: 'Customer Service' },
    { linkTitle: 'Help Center' },
    { linkTitle: 'Contact Us' },
    { linkTitle: 'Return an Item' },
    { linkTitle: 'Recall Information' },
    { linkTitle: 'Safety Data Sheets' }
  ],
  [
    { headerTitle: 'Corporate Info' },
    { linkTitle: 'Staples Business Advantage Overview' },
    { linkTitle: 'Industries Served' },
    { linkTitle: 'Blog' },
    { linkTitle: 'Our Brands' }
  ],
  [
    { headerTitle: 'Staples Corporate Solutions' },
    { linkTitle: 'Office Products' },
    { linkTitle: 'Technology Products' },
    { linkTitle: 'Furniture' },
    { linkTitle: 'Facilities' },
    { linkTitle: 'Promotional Products' }
  ],
  [
    { headerTitle: 'New Customer' },
    { linkTitle: 'Become a Customer' },
    { linkTitle: 'Govt. Customers' },
    { headerTitle: 'Memberships' },
    { linkTitle: 'Join the Staples Team' },
    { linkTitle: 'Careers' }
  ]
];

// accessibility image for FooterConnect component
const accessibilityImg = {
  src: `${IMAGE_URL}essential-accessibility.png`,
  alt: 'Essential Accessibility'
};

// social connect image for FooterConnect component
const socialImgs = [
  { src: `${IMAGE_URL}linkedin.png`, alt: 'LinkedIn' },
  { src: `${IMAGE_URL}twitter.png`, alt: 'Twitter' },
  { src: `${IMAGE_URL}facebook.png`, alt: 'Facebook' },
  { src: `${IMAGE_URL}youtube.png`, alt: 'Youtube' },
  { src: `${IMAGE_URL}insta.png`, alt: 'Instagram' }
];

// copyright text for footer copyright
const copyrightText =
  'Copyright &copy; 2001-2019 Corporate Express Canada, Inc., a Staples company. All rights reserved.';

// copyright links for footer copyright
const copyRightLinks = [
  'Terms & Conditions',
  ' | ',
  'Privacy Policy',
  ' | ',
  'Staples Bussiness Advantage'
];

// export all data
export {
  footerLinkStack,
  accessibilityImg,
  socialImgs,
  copyRightLinks,
  copyrightText
};
