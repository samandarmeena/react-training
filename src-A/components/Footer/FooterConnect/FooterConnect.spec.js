import React from 'react';
import { mount } from 'enzyme';
import FooterConnect from './FooterConnect';
import { accessibilityImg, socialImgs } from '../../../dataStore/data';

describe('FooterLink component', () => {
  let wrapper = '';
  beforeAll(() => {
    wrapper = mount(
      <FooterConnect
        accessibilityImg={accessibilityImg}
        socialImgs={socialImgs}
      />
    );
  });

  it('component rendered without an error', () => {
    expect(wrapper).toBeTruthy();
  });

  it('Text is rendering properly', () => {
    expect(
      wrapper
        .find('Text')
        .find('span')
        .text()
    ).toBe('Join us on');
  });

  it('Check connectivity links renders properly', () => {
    expect(wrapper.find('.footerConnectLinks').length).toBe(1);
  });

  it('Accessibility image is rendering properly', () => {
    expect(
      wrapper
        .find('.footerConnectLinks')
        .find('.footerConnectRight')
        .find('Image').length
    ).toBe(1);
  });

  it('Social links are rendering properly', () => {
    expect(
      wrapper
        .find('.footerConnectLinks')
        .find('.footerConnectLeft')
        .find('Image').length
    ).toBe(5);
    expect(
      wrapper
        .find('.footerConnectLinks')
        .find('.footerConnectLeft')
        .find('Image')
        .at(4)
        .prop('alt')
    ).toBe('Instagram');
  });

  it('Snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
