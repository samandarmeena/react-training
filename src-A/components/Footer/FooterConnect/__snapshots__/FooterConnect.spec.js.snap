// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLink component Snapshot testing 1`] = `
<FooterConnect
  accessibilityImg={
    Object {
      "alt": "Essential Accessibility",
      "src": "assets/images/essential-accessibility.png",
    }
  }
  socialImgs={
    Array [
      Object {
        "alt": "LinkedIn",
        "src": "assets/images/linkedin.png",
      },
      Object {
        "alt": "Twitter",
        "src": "assets/images/twitter.png",
      },
      Object {
        "alt": "Facebook",
        "src": "assets/images/facebook.png",
      },
      Object {
        "alt": "Youtube",
        "src": "assets/images/youtube.png",
      },
      Object {
        "alt": "Instagram",
        "src": "assets/images/insta.png",
      },
    ]
  }
>
  <div
    className="footerConnectTitle"
  >
    <Text
      isStrong={false}
      text="Join us on"
    >
      <span
        className=""
      >
        Join us on
      </span>
    </Text>
  </div>
  <div
    className="footerConnectLinks"
  >
    <div
      className="footerConnectLeft"
    >
      <span
        key="0"
      >
        <Image
          alt="LinkedIn"
          src="assets/images/linkedin.png"
        >
          <img
            alt="LinkedIn"
            src="assets/images/linkedin.png"
          />
        </Image>
      </span>
      <span
        key="1"
      >
        <Image
          alt="Twitter"
          src="assets/images/twitter.png"
        >
          <img
            alt="Twitter"
            src="assets/images/twitter.png"
          />
        </Image>
      </span>
      <span
        key="2"
      >
        <Image
          alt="Facebook"
          src="assets/images/facebook.png"
        >
          <img
            alt="Facebook"
            src="assets/images/facebook.png"
          />
        </Image>
      </span>
      <span
        key="3"
      >
        <Image
          alt="Youtube"
          src="assets/images/youtube.png"
        >
          <img
            alt="Youtube"
            src="assets/images/youtube.png"
          />
        </Image>
      </span>
      <span
        key="4"
      >
        <Image
          alt="Instagram"
          src="assets/images/insta.png"
        >
          <img
            alt="Instagram"
            src="assets/images/insta.png"
          />
        </Image>
      </span>
    </div>
    <div
      className="footerConnectRight"
    >
      <Image
        alt="Essential Accessibility"
        src="assets/images/essential-accessibility.png"
      >
        <img
          alt="Essential Accessibility"
          src="assets/images/essential-accessibility.png"
        />
      </Image>
    </div>
  </div>
</FooterConnect>
`;
